module/VuFindSearch/src/VuFindSearch/Backend/ProQuestFSG/Backend.php

```diff
@@ -0,0 +1,170 @@
+ * PHP version 8
+ *
+ * Copyright (C) Villanova University 2024.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+
+namespace VuFindSearch\Backend\ProQuestFSG;
+
+use VuFindSearch\Backend\AbstractBackend;
+use VuFindSearch\ParamBag;
+use VuFindSearch\Query\AbstractQuery;
+use VuFindSearch\Response\RecordCollectionFactoryInterface;
+use VuFindSearch\Response\RecordCollectionInterface;
+
+/**
+ * ProQuest Federated Search Gateway backend.
+ *
+ * @category VuFind
+ * @package  Search
+ * @author   David Maus <maus@hab.de>
+ * @author   Maccabee Levine <msl321@lehigh.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org
+ */
+class Backend extends AbstractBackend
+{
+    /**
+     * Connector.
+     *
+     * @var Connector
+     */
+    protected $connector;
+
+    /**
+     * Query builder.
+     *
+     * @var QueryBuilder
+     */
+    protected $queryBuilder = null;
+
+    /**
+     * Constructor.
+     *
+     * @param Connector                        $connector ProQuestFSG connector
+     * @param RecordCollectionFactoryInterface $factory   Record collection factory
+     * (null for default)
+     *
+     * @return void
+     */
+    public function __construct(
+        Connector $connector,
+        RecordCollectionFactoryInterface $factory = null
+    ) {
+        if (null !== $factory) {
+            $this->setRecordCollectionFactory($factory);
+        }
+        $this->connector    = $connector;
+        $this->identifier   = 'ProQuestFSG';
+    }
+
+    /**
+     * Perform a search and return record collection.
+     *
+     * @param AbstractQuery $query  Search query
+     * @param int           $offset Search offset
+     * @param int           $limit  Search limit
+     * @param ParamBag      $params Search backend parameters
+     *
+     * @return RecordCollectionInterface
+     */
+    public function search(
+        AbstractQuery $query,
+        $offset,
+        $limit,
+        ParamBag $params = null
+    ) {
+        if (null === $params) {
+            $params = new ParamBag();
+        }
```

Comment:
I _think_ you could simplify this to:

```suggestion
        $params ??= new ParamBag();
```